### PR TITLE
Fixed NIST Visual Display pages

### LINKED ISSUE #82: NIST file_path does not keep count of starting points
This means that only one of the starting points has a visual display page.

### PR DESCRIPTION
**Pull request by bb511**
#### Description of Work

Fixes #82 
Visual display pages are now produced for each starting position and the links are correct in the main tables.

#### Testing Instructions

1. Run `example_runScript.py` in the mantidpython console.
2. Check the links in the NIST tables and make sure that they link to the correct pages.

### CODE REVIEW

**Comment by Anders-Markvardsen** [comment]
merge conflict

**Comment by Anders-Markvardsen** [comment]
```
---------------------------------------------------------------------------
IOError                                   Traceback (most recent call last)
C:\Midlertiddig\fitbenchmarking\example_scripts\example_runScripts.py in <module>()
    100                     group_name=group_suffix_names[idx],
    101                     use_errors=use_errors, color_scale=color_scale,
--> 102                     results_dir=results_dir)

C:\Midlertiddig\fitbenchmarking\fitbenchmarking\results_output.py in save_results_tables(minimizers, results_per_test, group_name, use_errors, color_scale, results_dir)
     71     tables_dir = make_result_tables_directory(results_dir, group_name)
     72     linked_problems = build_indiv_linked_problems(results_per_test, group_name,
---> 73                                                   results_dir)
     74
     75     accuracy_tbl, runtime_tbl = \

C:\Midlertiddig\fitbenchmarking\fitbenchmarking\results_output.py in build_indiv_linked_problems(results_per_test, group_name, results_dir)
    144         name = '`' + name_index + ' ' + \
    145                build_visual_display_page(prob_results, group_name, results_dir,
--> 146                                          count)
    147         linked_problems.append(name)
    148

C:\Midlertiddig\fitbenchmarking\fitbenchmarking\results_output.py in build_visual_display_page(prob_results, group_name, results_dir, count)
    258                                figure_fit, fit_function_details_table,
    259                                gb.minimizer, see_also_link)
--> 260     save_VDpages(rst_text, problem_name, file_path)
    261
    262     return rst_link

C:\Midlertiddig\fitbenchmarking\fitbenchmarking\results_output.py in save_VDpages(rst_text, prob_name, file_path)
    348
    349     html = publish_string(rst_text, writer_name='html')
--> 350     with open(file_path + '.' + FILENAME_EXT_TXT, 'w') as visual_rst:
    351         print(html, file=visual_rst)
    352         logger.info('Saved {prob_name}.{extension} to {working_directory}'.

IOError: [Errno 2] No such file or directory: 'C:\\Midlertiddig\\fitbenchmarking\\example_scripts\\results\\neutron\\VDPages\\neutron_data_enginx_193749_calibration_spectrum_651_peak_19_1.txt'
```

**Comment by bb511** [comment]
@Anders-Markvardsen 
Sorry, tested it on my machine and it produces the expected results. Did you remote update and pull?

**Comment by Anders-Markvardsen** [comment]
No worries - my mistakes - I needed to restart the console. Test and NIST pages looks good including the line to external page at the bottom

File: fitbenchmarking/results_output.py [hunk 1]
```diff
@@ -173,36 +176,34 @@ def setup_neutron_VDpage_misc(function_def, results_dir):
     return support_pages_dir, details_table, see_also_link
 
 
-def setup_VDpage_misc(group_name, problem_name, res_obj, results_dir):
+def setup_VDpage_misc(group_name, problem_name, res_obj, results_dir, count):
     """
     Setup miscellaneous data for visual display page.
```
Comment (Anders-Markvardsen) [inline fitbenchmarking/results_output.py]: Could you add function @param comments please? For example here not immediately obvious what count is

File: fitbenchmarking/results_output.py [hunk 2]
```diff
@@ -173,36 +176,34 @@ def setup_neutron_VDpage_misc(function_def, results_dir):
...
+    file_name = (group_name + '_' + problem_name + '_' + str(count)).lower()
     file_path = os.path.join(support_pages_dir, file_name)
 
-    return (support_pages_dir, file_path, fit_function_details_table,
-            see_also_link)
+    return support_pages_dir, file_path, fit_function_details_table, see_also_link
 
 
-def get_figures_path(support_pages_dir, problem_name):
+def get_figures_path(support_pages_dir, problem_name, count):
```
Comment (Anders-Markvardsen) [inline fitbenchmarking/results_output.py]: Same here please. There will be some repetition of @param comments which is ok

File: fitbenchmarking/results_output.py [hunk 3]
```diff
@@ -213,8 +214,7 @@ def get_figures_path(support_pages_dir, problem_name):
     return figure_data, figure_fit, figure_start
 
 
-
-def build_visual_display_page(prob_results, group_name, results_dir):
+def build_visual_display_page(prob_results, group_name, results_dir, count):
     """
     Builds a page containing details of the best fit for a problem.
     @param prob_results:: the list of results for a problem
```
Comment (Anders-Markvardsen) [inline fitbenchmarking/results_output.py]: @param for results_dir and count missing

**Comment by bb511** [comment]
I am almost done restructuring the whole project (since having a bunch of big .py files is not considered good practice as it is hard to read and understand what the script is doing - can confirm that this was the case when I started working on this). Therefore, I will add full function descriptions in #88 if that is ok.

**Comment by Anders-Markvardsen** [comment]
That sounds good

**Pull request merged**